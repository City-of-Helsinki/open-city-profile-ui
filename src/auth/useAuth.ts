import * as Sentry from '@sentry/react';
import i18n from 'i18next';
import { useOidcClient } from 'hds-react';

const origin = window.location.origin;

const NETWORK_ERROR = 'Network Error';

const useAuth = () => {
  const oidcClient = useOidcClient();
  const userManager = oidcClient.getUserManager();

  const logout = async (): Promise<void> => {
    await oidcClient.logout({
      extraQueryParams: { ui_locales: i18n.language },
    });
  };

  const changePassword = async (): Promise<void> => {
    let success = true;
    await userManager
      .signinRedirect({
        ui_locales: i18n.language,
        redirect_uri: `${origin}/password-change-callback`,
        extraQueryParams: {
          kc_action: 'UPDATE_PASSWORD',
        },
      })
      .catch((error) => {
        success = false;
        if (error.message !== NETWORK_ERROR) {
          Sentry.captureException(error);
        }
      });
    return success ? Promise.resolve() : Promise.reject();
  };

  const initiateTOTP = async (): Promise<void> => {
    let success = true;
    await userManager
      .signinRedirect({
        ui_locales: i18n.language,
        redirect_uri: `${origin}/otp-configuration-callback`, // otp-configuration-callback
        extraQueryParams: {
          kc_action: 'CONFIGURE_TOTP',
        },
      })
      .catch((error) => {
        success = false;
        if (error.message !== NETWORK_ERROR) {
          Sentry.captureException(error);
        }
      });
    return success ? Promise.resolve() : Promise.reject();
  };

  const disableTOTP = async (id: string | null): Promise<void> => {
    let success = true;

    await userManager
      .signinRedirect({
        ui_locales: i18n.language,
        redirect_uri: `${origin}/delete-credential-callback`, // otp-configuration-callback
        extraQueryParams: {
          kc_action: 'delete_credential:' + id,
        },
      })
      .catch((error) => {
        success = false;
        if (error.message !== NETWORK_ERROR) {
          Sentry.captureException(error);
        }
      });
    return success ? Promise.resolve() : Promise.reject();
  };

  return {
    isAuthenticated: oidcClient.isAuthenticated,
    login: oidcClient.login,
    logout,
    changePassword,
    initiateTOTP,
    disableTOTP,
  };
};

export default useAuth;
