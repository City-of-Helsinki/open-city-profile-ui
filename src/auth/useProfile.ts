import { User } from 'oidc-client-ts';
import React from 'react';
import { useOidcClient } from 'hds-react';

export const tunnistusSuomifiAMR = 'heltunnistussuomifi';

export type AMRStatic =
  | 'github'
  | 'google'
  | 'facebook'
  | 'yletunnus'
  | 'helsinkiAccount'
  | 'tunnistusSuomifi';

export interface Profile {
  amr: NonNullable<User['profile']['amr']>;
  auth_time: number;
  email: string;
  email_verified: boolean;
  family_name: string;
  given_name: string;
  name: string;
  nickname: string;
  sub: string;
}

export interface ProfileState {
  profile: Profile | null;
  loading: boolean;
  error: Error | null;
}

function getUserProfile(user: User | null): Profile | null {
  if (user === null || user.expired !== false) {
    return null;
  }
  const profile = { ...user.profile } as Profile;
  const amr = user.profile.amr;
  if (!amr) {
    profile.amr = [];
  } else if (typeof amr === 'string') {
    profile.amr = [amr];
  }
  return profile;
}

function useProfile(): ProfileState {
  const [profile, setProfile] = React.useState<Profile | null>(null);
  const [isLoading, setIsLoading] = React.useState<boolean>(false);
  const [error, setError] = React.useState<Error | null>(null);

  const { getUser } = useOidcClient();

  React.useEffect(() => {
    function setup() {
      setIsLoading(true);

      const user = getUser();

      if (user) {
        setProfile(getUserProfile(user));
      } else {
        setError(Error('User was not found'));
      }

      setIsLoading(false);
    }

    setup();
  }, [getUser]);

  return {
    profile,
    loading: isLoading,
    error,
  };
}

export default useProfile;
