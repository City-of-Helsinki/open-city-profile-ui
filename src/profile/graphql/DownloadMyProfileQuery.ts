import gql from 'graphql-tag';
export const DOWNLOAD_MY_PROFILE = gql`
  query DownloadMyProfile(
    $authorizationCode: String!
    $authorizationCodeKeycloak: String
  ) {
    downloadMyProfile(authorizationCode: $authorizationCode)
  }
`;
