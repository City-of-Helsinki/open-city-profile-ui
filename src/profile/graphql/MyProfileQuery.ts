/* eslint-disable max-len */
import gql from 'graphql-tag';

export const MY_PROFILE = gql`
  fragment MyProfileQueryVerifiedPersonalInformationPermanentForeignAddress on VerifiedPersonalInformationForeignAddressNode {
    streetAddress
    additionalAddress
    countryCode
  }

  fragment MyProfileQueryVerifiedPersonalInformationPermanentAddress on VerifiedPersonalInformationAddressNode {
    streetAddress
    postalCode
    postOffice
  }

  fragment MyProfileQueryVerifiedPersonalInformation on VerifiedPersonalInformationNode {
    firstName
    lastName
    givenName
    nationalIdentificationNumber
    municipalityOfResidence
    municipalityOfResidenceNumber
    permanentAddress {
      ...MyProfileQueryVerifiedPersonalInformationPermanentAddress
    }
    permanentForeignAddress {
      ...MyProfileQueryVerifiedPersonalInformationPermanentForeignAddress
    }
  }

  fragment MyProfileQueryPrimaryAddress on AddressNode {
    id
    primary
    address
    postalCode
    city
    countryCode
    addressType
  }

  fragment MyProfileQueryPrimaryEmail on EmailNode {
    id
    email
    primary
    emailType
  }

  fragment MyProfileQueryPrimaryPhone on PhoneNode {
    id
    phone
    primary
    phoneType
  }

  fragment MyProfileQueryAddressesEdgesNode on AddressNode {
    primary
    id
    address
    postalCode
    city
    countryCode
    addressType
  }

  fragment MyProfileQueryEmailsEdgesNode on EmailNode {
    primary
    id
    email
    emailType
  }

  fragment MyProfileQueryPhonesEdgesNode on PhoneNode {
    primary
    id
    phone
    phoneType
  }

  fragment MyLoginMethodNode on LoginMethodNode {
    method
    createdAt
    credentialId
    userLabel
  }

  fragment MyProfileQueryAddressesEdges on AddressNodeEdge {
    node {
      ...MyProfileQueryAddressesEdgesNode
    }
  }

  fragment MyProfileQueryEmailsEdges on EmailNodeEdge {
    node {
      ...MyProfileQueryEmailsEdgesNode
    }
  }

  fragment MyProfileQueryPhonesEdges on PhoneNodeEdge {
    node {
      ...MyProfileQueryPhonesEdgesNode
    }
  }

  fragment MyProfileQueryAddresses on AddressNodeConnection {
    edges {
      ...MyProfileQueryAddressesEdges
    }
  }

  fragment MyProfileQueryEmails on EmailNodeConnection {
    edges {
      ...MyProfileQueryEmailsEdges
    }
  }

  fragment MyProfileQueryPhones on PhoneNodeConnection {
    edges {
      ...MyProfileQueryPhonesEdges
    }
  }

  fragment MyProfileQuery on ProfileNode {
    id
    firstName
    lastName
    nickname
    language
    availableLoginMethods {
      ...MyLoginMethodNode
    }
    primaryAddress {
      ...MyProfileQueryPrimaryAddress
    }
    addresses {
      ...MyProfileQueryAddresses
    }
    primaryEmail {
      ...MyProfileQueryPrimaryEmail
    }
    emails {
      ...MyProfileQueryEmails
    }
    primaryPhone {
      ...MyProfileQueryPrimaryPhone
    }
    phones {
      ...MyProfileQueryPhones
    }
    verifiedPersonalInformation {
      ...MyProfileQueryVerifiedPersonalInformation
    }
  }

  query MyProfile {
    myProfile {
      ...MyProfileQuery
    }
  }
`;
