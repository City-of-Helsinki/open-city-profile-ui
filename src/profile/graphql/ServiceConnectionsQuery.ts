import gql from 'graphql-tag';

export const SERVICE_CONNECTIONS = gql`
  fragment ServiceConnectionsQueryEdgesNodeServiceAllowedDataFieldsEdgesNode on AllowedDataFieldNode {
    fieldName
    label(language: $language)
  }

  fragment ServiceConnectionsQueryEdgesNodeServiceAllowedDataFieldsEdges on AllowedDataFieldNodeEdge {
    node {
      ...ServiceConnectionsQueryEdgesNodeServiceAllowedDataFieldsEdgesNode
    }
  }

  fragment ServiceConnectionsQueryEdgesNodeService on ServiceNode {
    name
    title(language: $language)
    description(language: $language)
    allowedDataFields {
      edges {
        ...ServiceConnectionsQueryEdgesNodeServiceAllowedDataFieldsEdges
      }
    }
  }

  fragment ServiceConnectionsQueryEdgesNode on ServiceConnectionType {
    service {
      ...ServiceConnectionsQueryEdgesNodeService
    }
    createdAt
  }

  query ServiceConnections($language: TranslationLanguage!) {
    myProfile {
      id
      serviceConnections {
        edges {
          node {
            ...ServiceConnectionsQueryEdgesNode
          }
        }
      }
    }
  }
`;
