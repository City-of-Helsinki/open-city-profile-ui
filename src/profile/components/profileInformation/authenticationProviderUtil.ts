import { Amr } from 'hds-react';

import { AMRStatic, tunnistusSuomifiAMR } from '../../../auth/useProfile';
import config from '../../../config';
import { LoginMethodType, ProfileRoot } from '../../../graphql/typings';
import { MyLoginMethodNodeFragment } from '../../../graphql/generatedTypes';

function getAmrFromArray(amrArray: Amr | undefined): string | undefined {
  return amrArray && Array.isArray(amrArray) ? amrArray[0] : '';
}

export function getAmrStatic(amrArray: Amr | undefined): AMRStatic | null {
  const amr = getAmrFromArray(amrArray);

  // If amr designates helsinki account, switch the value into a static
  // value. This setup allows the amr for Helsinki account to be
  // configured, while allowing a static reference we can use for
  // translations for instance.
  if (amr === config.helsinkiAccountAMR) {
    return 'helsinkiAccount';
  }
  if (amr === tunnistusSuomifiAMR) {
    return 'tunnistusSuomifi';
  }

  // TODO: Should be removed soon
  if (amr === 'github' || amr === 'google' || amr === 'facebook' || amr === 'yletunnus') {
    return amr;
  }

  // If amr doesn't match any of our expectations, return null.
  return null;
}

export function hasTunnistusSuomiFiAmr(amrArray: Amr | undefined): boolean {
  return getAmrFromArray(amrArray) === tunnistusSuomifiAMR;
}

export function hasHelsinkiAccountAMR(amrArray: Amr | undefined): boolean {
  return getAmrFromArray(amrArray) === config.helsinkiAccountAMR;
}

export function hasPasswordLogin(data: ProfileRoot | undefined): boolean {
  return (
    data?.myProfile?.availableLoginMethods?.some((item) => item && item.method === LoginMethodType.PASSWORD) ?? false
  );
}

export function getMFALoginMethod(data: ProfileRoot | undefined): MyLoginMethodNodeFragment | undefined {
  const loginMethod = data?.myProfile?.availableLoginMethods?.find(
    (item) => item && item.method === LoginMethodType.OTP,
  );

  return loginMethod || undefined;
}

export function formatDate(dateString: string): string | null {
  const date = new Date(dateString);
  // Extract day, month, and year
  const day = String(date.getUTCDate()).padStart(2, '0');
  const month = String(date.getUTCMonth() + 1);
  const year = date.getUTCFullYear();
  // Format to DD.MM.YYYY
  return `${day}.${month}.${year}`;
}
