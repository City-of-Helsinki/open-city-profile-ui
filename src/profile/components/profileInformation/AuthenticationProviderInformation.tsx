import React, { Fragment, useContext, useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { Button, useOidcClient } from 'hds-react';
import classNames from 'classnames';

import { getAmrStatic, hasPasswordLogin } from './authenticationProviderUtil';
import OtpInformation from './OtpInformation';
import ProfileSection from '../../../common/profileSection/ProfileSection';
import commonFormStyles from '../../../common/cssHelpers/form.module.css';
import { ProfileContext } from '../../context/ProfileContext';
import useNotificationContent from '../editingNotifications/useNotificationContent';
import EditingNotifications from '../editingNotifications/EditingNotifications';
import useAuth from '../../../auth/useAuth';
import config from '../../../config';

function AuthenticationProviderInformation(): React.ReactElement | null {
  const { t } = useTranslation();
  const notificationRef = useRef<HTMLDivElement>(null);

  const { data, passwordUpdateState, setPasswordUpdateState } = useContext(ProfileContext);

  const hasPassword = hasPasswordLogin(data);
  const { getAmr } = useOidcClient();
  const amr = getAmrStatic(getAmr());
  const showSuccess = passwordUpdateState;
  const { content, setSuccessMessage } = useNotificationContent();
  const { changePassword } = useAuth();

  useEffect(() => {
    if (showSuccess) {
      setSuccessMessage('save');
      setPasswordUpdateState(false);

      // Scrolling needs timeout because the notification is not yet rendered
      setTimeout(() => {
        if (notificationRef.current) {
          notificationRef.current.scrollIntoView({
            behavior: 'auto',
            block: 'center',
          });
        }
      }, 0);
    }
  }, [showSuccess, setPasswordUpdateState, setSuccessMessage]);

  if (!amr) {
    return null;
  }

  const authenticationMethodReferenceName = t(`identityProvider.${amr}`);

  const flexBoxColumns = 'responsive-flex-box-columns-rows';

  return (
    <ProfileSection>
      <div className={commonFormStyles['flex-box-columns']}>
        <div className={classNames(commonFormStyles['editor-description-container'])}>
          <h2>{t('profileInformation.loginAndAuthentication')}</h2>
          <span>{authenticationMethodReferenceName}</span>
        </div>

        {hasPassword && (
          <Fragment>
            <div className={classNames(commonFormStyles[flexBoxColumns], commonFormStyles['password-container'])}>
              <div
                className={classNames(
                  commonFormStyles['editor-title-and-value'],
                  commonFormStyles['responsive-flex-box-columns-rows'],
                )}
              >
                <h3 className={commonFormStyles['subtitle-size']}>{t('profileInformation.password')}</h3>
              </div>
              <div className={commonFormStyles['edit-buttons']}>
                <div className={commonFormStyles['edit-buttons-container']}>
                  <Button
                    data-testid={'change-password-button'}
                    onClick={() => {
                      changePassword();
                    }}
                  >
                    {t('profileInformation.changePassword')}
                  </Button>
                </div>
              </div>
            </div>
            <EditingNotifications ref={notificationRef} content={content} dataType={'password'} bottomSpacing />

            {config.mfa && <OtpInformation />}
          </Fragment>
        )}
      </div>
    </ProfileSection>
  );
}

export default AuthenticationProviderInformation;
