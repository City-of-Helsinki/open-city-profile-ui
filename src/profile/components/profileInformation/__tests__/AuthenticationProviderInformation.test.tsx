import React from 'react';
import { render } from '@testing-library/react';

import AuthenticationProviderInformation from '../AuthenticationProviderInformation';
import { mockProfileCreator } from '../../../../common/test/userMocking';
import TestLoginProvider from '../../../../common/test/TestLoginProvider';
import * as useProfile from '../../../../auth/useProfile';
import * as authenticationProviderUtil from '../authenticationProviderUtil';
import { LoginMethodType } from '../../../../graphql/typings';
import { MyLoginMethodNodeFragment } from '../../../../graphql/generatedTypes';

const helsinkiAccountAMR = 'helsinki_tunnus-test';

vi.spyOn(useProfile, 'default').mockImplementation(
  () =>
    (({
      profile: mockProfileCreator(),
    } as unknown) as useProfile.ProfileState)
);

const mockLoginMethodNodeFragment: MyLoginMethodNodeFragment = {
  createdAt: '2023-01-01T00:00:00Z',
  method: LoginMethodType.OTP,
  __typename: 'LoginMethodNode',
  credentialId: '3837373',
  userLabel: 'foo',
};

describe('<AuthenticationProviderInformation /> ', () => {
  const defaultProps = {};

  const getWrapper = () =>
    render(
      <TestLoginProvider>
        <AuthenticationProviderInformation {...defaultProps} />
      </TestLoginProvider>
    );

  describe('renders correctly when AMR is helsinkiAccountAMR', () => {
    beforeAll(() => {
      window._env_.REACT_APP_HELSINKI_ACCOUNT_AMR = helsinkiAccountAMR;
    });

    afterAll(() => {
      window._env_.REACT_APP_HELSINKI_ACCOUNT_AMR = 'helsinki_tunnus';
    });

    it('should render helsinki account information as expected based on config', () => {
      vi.spyOn(authenticationProviderUtil, 'getAmrStatic').mockImplementation(
        () => 'helsinkiAccount' as useProfile.AMRStatic
      );

      const { container } = getWrapper();
      expect(container).toMatchSnapshot();
    });
  });
  describe('renders correctly when AMR is tunnistusSuomifiAMR', () => {
    it('should render suomi.fi information as expected based on config', () => {
      vi.spyOn(authenticationProviderUtil, 'getAmrStatic').mockImplementation(
        () => 'tunnistusSuomifi' as useProfile.AMRStatic
      );

      const { container } = getWrapper();
      expect(container).toMatchSnapshot();
    });
  });

  describe('renders correctly according to Login method', () => {
    it('should render change password button when password is used', () => {
      vi.spyOn(
        authenticationProviderUtil,
        'hasPasswordLogin'
      ).mockImplementation(() => true);

      const { container, getByTestId } = getWrapper();
      expect(getByTestId('change-password-button')).toBeVisible();
      expect(container).toMatchSnapshot();
    });

    it('should not render change password button when password is not used', () => {
      vi.spyOn(
        authenticationProviderUtil,
        'hasPasswordLogin'
      ).mockImplementation(() => false);

      const { container, queryByTestId } = getWrapper();
      expect(queryByTestId('change-password-button')).not.toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  describe('renders correctly according to MFA setting', () => {
    it('should render enable MFA button', () => {
      vi.spyOn(
        authenticationProviderUtil,
        'hasPasswordLogin'
      ).mockImplementation(() => true);

      vi.spyOn(
        authenticationProviderUtil,
        'getMFALoginMethod'
      ).mockImplementation(() => undefined);

      const { container, queryByTestId } = getWrapper();
      expect(queryByTestId('enable-totp-button')).toBeVisible();
      expect(queryByTestId('disable-totp-button')).not.toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it('should render "disable MFA button" when MFA is already configured', () => {
      vi.spyOn(
        authenticationProviderUtil,
        'hasPasswordLogin'
      ).mockImplementation(() => true);

      vi.spyOn(
        authenticationProviderUtil,
        'getMFALoginMethod'
      ).mockImplementation(() => mockLoginMethodNodeFragment);

      const { container, queryByTestId } = getWrapper();
      expect(queryByTestId('enable-totp-button')).not.toBeInTheDocument();
      expect(queryByTestId('disable-totp-button')).toBeVisible();
      expect(container).toMatchSnapshot();
    });
  });
});
