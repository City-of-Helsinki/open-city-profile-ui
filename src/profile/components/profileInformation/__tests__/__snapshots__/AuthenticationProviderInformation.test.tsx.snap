// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AuthenticationProviderInformation />  > renders correctly according to Login method > should not render change password button when password is not used 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.tunnistusSuomifi
        </span>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AuthenticationProviderInformation />  > renders correctly according to Login method > should render change password button when password is used 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.tunnistusSuomifi
        </span>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <h3
            class="subtitle-size"
          >
            profileInformation.password
          </h3>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              data-testid="change-password-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                profileInformation.changePassword
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <div>
            <h3
              class="subtitle-size"
            >
              mfa.title
            </h3>
            <span
              class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_statusLabelWithIcon__5lKVk status-label_hds-status-label--with-icon__3bHf6"
            >
              <span
                aria-hidden="true"
                class="StatusLabel-module_statusLabelIcon__2SAou status-label_hds-status-label-icon__3K58a"
              >
                <svg
                  aria-hidden="true"
                  aria-label="alert-circle"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4ZM13 16V18H11V16H13ZM13 6V14H11V6H13Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              mfa.disabled
            </span>
          </div>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              data-testid="enable-totp-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                mfa.enable
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AuthenticationProviderInformation />  > renders correctly according to MFA setting > should render "disable MFA button" when MFA is already configured 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.tunnistusSuomifi
        </span>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <h3
            class="subtitle-size"
          >
            profileInformation.password
          </h3>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              data-testid="change-password-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                profileInformation.changePassword
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <div>
            <h3
              class="subtitle-size"
            >
              mfa.title
            </h3>
            <span
              class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_success__2BfdX status-label_hds-status-label--success__3jiST StatusLabel-module_statusLabelWithIcon__5lKVk status-label_hds-status-label--with-icon__3bHf6"
            >
              <span
                aria-hidden="true"
                class="StatusLabel-module_statusLabelIcon__2SAou status-label_hds-status-label-icon__3K58a"
              >
                <svg
                  aria-hidden="true"
                  aria-label="check-circle"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4ZM16.5 8L18 9.5L10.5 17L6 12.5L7.5 11L10.5 14L16.5 8Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              mfa.dateEnabled
               
              01.1.2023
            </span>
          </div>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS"
              data-testid="disable-totp-button"
              type="button"
            >
              <div
                aria-hidden="true"
                class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
              >
                <svg
                  aria-hidden="true"
                  aria-label="cross-circle"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4ZM15 7.5L16.5 9L13.5 12L16.5 15L15 16.5L12 13.5L9 16.5L7.5 15L10.5 12L7.5 9L9 7.5L12 10.5L15 7.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                mfa.disable
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AuthenticationProviderInformation />  > renders correctly according to MFA setting > should render enable MFA button 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.tunnistusSuomifi
        </span>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <h3
            class="subtitle-size"
          >
            profileInformation.password
          </h3>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              data-testid="change-password-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                profileInformation.changePassword
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="responsive-flex-box-columns-rows password-container"
      >
        <div
          class="editor-title-and-value responsive-flex-box-columns-rows"
        >
          <div>
            <h3
              class="subtitle-size"
            >
              mfa.title
            </h3>
            <span
              class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_statusLabelWithIcon__5lKVk status-label_hds-status-label--with-icon__3bHf6"
            >
              <span
                aria-hidden="true"
                class="StatusLabel-module_statusLabelIcon__2SAou status-label_hds-status-label-icon__3K58a"
              >
                <svg
                  aria-hidden="true"
                  aria-label="alert-circle"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4ZM13 16V18H11V16H13ZM13 6V14H11V6H13Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              mfa.disabled
            </span>
          </div>
        </div>
        <div
          class="edit-buttons"
        >
          <div
            class="edit-buttons-container"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              data-testid="enable-totp-button"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              >
                mfa.enable
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AuthenticationProviderInformation />  > renders correctly when AMR is helsinkiAccountAMR > should render helsinki account information as expected based on config 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.helsinkiAccount
        </span>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AuthenticationProviderInformation />  > renders correctly when AMR is tunnistusSuomifiAMR > should render suomi.fi information as expected based on config 1`] = `
<div>
  <section
    class="content-box profile-section"
  >
    <div
      class="flex-box-columns"
    >
      <div
        class="editor-description-container"
      >
        <h2>
          profileInformation.loginAndAuthentication
        </h2>
        <span>
          identityProvider.tunnistusSuomifi
        </span>
      </div>
    </div>
  </section>
</div>
`;
